packages/client/src/serialize.js

```diff
@@ -23,7 +23,12 @@ let mapTree = (fn, tree) =>
 
 export default (tree, types, { search } = {}) => {
   let onSerialize = (node) =>
-    runTypeFunctionOrDefault(_.identity, types, 'onSerialize', node, {})
+    runTypeFunctionOrDefault(_.identity, types, 'onSerialize', node, {
+      extend: _.identity,
+      snapshot: _.identity,
+      initObject: _.identity,
+      log: _.identity,
```

Comment:
We should either thread the right props through or keep this undefined. Calling `_.identity` when you think it's `extend` or `snapshot` is definitely more surprising than just having it explode as undefined.

Reply:
Changed to empty object